registry/registry.go

```diff
@@ -116,3 +152,64 @@ func (c *Client) handleSingleArchImage(desc *remote.Descriptor, info *Image) err
 
 	return nil
 }
+
+type ImageFetchResult struct {
+	Reference name.Reference
+	Results   map[string]Image // registry name -> image info
+}
+
+type RegistryClient interface {
+	Image(ctx context.Context, ref name.Reference) (Image, error)
+}
+
+type MultiRegistryFetcher struct {
+	registries map[string]RegistryClient
+}
+
+func NewMultiRegistryFetcher(registries map[string]RegistryClient) *MultiRegistryFetcher {
+	return &MultiRegistryFetcher{
+		registries: registries,
+	}
+}
+
+func (f *MultiRegistryFetcher) FetchImages(ctx context.Context, refs []name.Reference) (<-chan ImageFetchResult, <-chan error) {
```

Comment:
This name stutters. Can be called `Images`. Can you update the receiver variable to be `m` also?